src/Umbraco.Web/PropertyEditors/ValueConverters/BlockListPropertyValueConverter.cs

```diff
@@ -120,7 +122,18 @@ public override object ConvertIntermediateToObject(IPublishedElement owner, IPub
                             settingsData = null;
                     }
 
-                    var layoutType = typeof(BlockListItem<,>).MakeGenericType(contentData.GetType(), settingsData?.GetType() ?? typeof(IPublishedElement));
+                    // Get settings type from configuration
+                    Type settingsType;
+                    if (blockConfig.SettingsElementTypeKey.HasValue)
+                    {
+                        settingsType = _blockConverter.GetModelType(blockConfig.SettingsElementTypeKey.Value);
+                    }
+                    else
+                    {
+                        settingsType = typeof(IPublishedElement);
+                    }
+
+                    var layoutType = typeof(BlockListItem<,>).MakeGenericType(contentData.GetType(), settingsType);
                     var layoutRef = (BlockListItem)Activator.CreateInstance(layoutType, contentGuidUdi, contentData, settingGuidUdi, settingsData);
```

Comment:
This should use compiled lambda expressions, as that's _way_ faster than `Activator.CreateInstance()`, especially with the parameterized constructor!

If we cache on `contentTypeKey` and `blockConfig.SettingsElementTypeKey`, getting the settings type could be included in the cache, which reduces calls to `GetModelType()` 👍

Any pointers on how this is best achieved?